torch/onnx/_internal/fx/registration.py

```diff
@@ -2,177 +2,64 @@
 
 from __future__ import annotations
 
-import warnings
-from typing import (
-    Callable,
-    Collection,
-    Dict,
-    Generic,
-    Optional,
-    Set,
-    TYPE_CHECKING,
-    TypeVar,
-)
+import dataclasses
+from collections import defaultdict
+from typing import Dict, Optional, Set, TYPE_CHECKING, Union
 
 from torch.onnx._internal import _beartype
 
 # We can only import onnx from this module in a type-checking context to ensure that
 # 'import torch.onnx' continues to work without having 'onnx' installed. We fully
 # 'import onnx' inside of dynamo_export (by way of _assert_dependencies).
 if TYPE_CHECKING:
+    import onnxscript  # type: ignore[import]
     from onnxscript.function_libs.torch_lib import registration  # type: ignore[import]
 
 OpsetVersion = int
-_K = TypeVar("_K")
-_V = TypeVar("_V")
 
 
-class MergeDict(Generic[_K, _V], Collection[_K]):
-    """
-    A dictionary that merges built-in and custom symbolic functions.
+@dataclasses.dataclass(frozen=True, eq=True)
+class SymbolicFunction:
+    """A wrapper of onnx-script function.
 
-    It supports adding and removing built-in symbolic functions with custom ones.
+    op_name: The qualified name of the function. In the form of 'domain::op'.
```

Comment:
Will there be the overload name here too? Is it from the torchlib registry?

Reply:
There should. I should update this guy too. There ae two ways to name here. One directly from torchlib, and the other one we get domain, op_name, and overload from users, and then combine them to op_name. Maybe we should have a common name on the whole thing. Maybe something like: operator_name = <domain>::<op_name>.<overload>